source/03-components/accordion/accordion.es6.js

```diff
@@ -0,0 +1,170 @@
+          document.getElementById(button.getAttribute('aria-controls')),
+          ACCORDION_SPEED
+        );
+      }
+    };
+
+    // Accessible Accordion Functionality
+    // Based off example work from W3 best aria practices
+    // https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html
+    accordions.forEach(accordion => {
+      // Allow for multiple accordion sections to be expanded at the same time
+      const allowMultiple = accordion.hasAttribute('data-allow-multiple');
+      // Allow for each toggle to both open and close individually
+      const allowToggle = allowMultiple
+        ? true
+        : accordion.hasAttribute('data-allow-toggle');
+
+      // Create the array of toggle elements for the accordion group
+      const triggers = Array.prototype.slice.call(
+        accordion.querySelectorAll(`.${ACCORDION_TOGGLE_CLASS}`)
+      );
+
+      accordion.addEventListener('click', event => {
+        // Set target differently depending on click vs. keydown
+        // because the <span> inside <button> screws things up
+        if (
+          event.target.classList.contains(ACCORDION_TOGGLE_CLASS) ||
+          event.target.parentElement.classList.contains(
+            ACCORDION_TOGGLE_CLASS
+          )
+        ) {
+          let target;
+          // Set target based on click or keydown
+          if (event.target.classList.contains(ACCORDION_TOGGLE_CLASS)) {
+            target = event.target;
+          } else {
+            target = event.target.parentElement;
+          }
+          // Check if the current toggle is expanded.
+          const isExpanded = target.getAttribute('aria-expanded') === 'true';
+          const active = accordion.querySelector('[aria-expanded="true"]');
+
+          // without allowMultiple, close the open accordion
+          if (!allowMultiple && active && active !== target) {
+            closeAccordion(
+              document.getElementById(active.getAttribute('aria-controls')),
+              active,
+              allowToggle,
+              true
+            );
+          }
+
+          if (!isExpanded) {
+            openAccordion(
+              document.getElementById(target.getAttribute('aria-controls')),
+              target,
+              allowToggle,
+              !allowMultiple
+            );
+          } else if (allowToggle && isExpanded) {
+            closeAccordion(
+              document.getElementById(target.getAttribute('aria-controls')),
+              target,
+              allowToggle,
+              !allowMultiple
+            );
+          }
+
+          event.preventDefault();
+        }
+      });
+
+      // Bind keyboard behaviors on the main accordion container
+      accordion.addEventListener('keydown', event => {
+        const currentTarget = event.target;
+        const key = event.which.toString();
+
+        // 33 = Page Up, 34 = Page Down
+        const ctrlModifier = event.ctrlKey && key.match(/33|34/);
```

Comment:
Instead of hard-coding the keycodes, we should pull in constants from `source/00-config/_KEYCODE.es6.js`.

Reply:
Replaced key board matches